docs/source/etcd-architecture.rst

```diff
@@ -0,0 +1,238 @@
+Calico etcd-based architecture
```

Comment:
Yes, we ought to decide on the overall messaging approach.
If we look at this document in six month's time, it will just describe the Calico Architecture.  The fact that we moved from 0MQ to etcd will be lost int he mists of time.  Consequently, I think I'd rather see this just called "Calico Architecture".
However, in introducing the "new" architecture, we'll probably want to make it clear that we're moving to use etcd.  That may mean that we:

-  use the blog and mailing list to advertise our "upcoming architecture changes" with a heavy reference to etcd
-  at that point have this document sit in the "Future" section of the docs with a clear "Coming soon" banner and have it referenced from the current 0MQ architecture and API docs
-  once we actually release the etcd code into master, we move these new docs to replace the current 0MQ ones.

Alternatively, we could leave these docs in a branch (still, potentially, blog their existence) and then just merge them all into master when the code is ready.

I'm interested in opinions here, but we'll probably want to make a proposal to (and get sign off from) the CCDT.